core/src/test/java/org/elasticsearch/tribe/TribeServiceTests.java

```diff
@@ -96,4 +109,128 @@ public void testPassthroughSettings() {
         assertEquals("7.7.7.7", clientSettings.get("transport.bind_host"));
         assertEquals("8.8.8.8", clientSettings.get("transport.publish_host"));
     }
+
+    public void testMergeCustomMetaDataSimple() {
+        Map<String, MetaData.Custom> mergedCustoms =
+                TribeService.mergeChangedCustomMetaData(Collections.singleton(MergableCustomMetaData1.TYPE),
+                        s -> Collections.singletonList(new MergableCustomMetaData1("data1")));
+        TestCustomMetaData mergedCustom = (TestCustomMetaData) mergedCustoms.get(MergableCustomMetaData1.TYPE);
+        assertThat(mergedCustom, instanceOf(MergableCustomMetaData1.class));
+        assertNotNull(mergedCustom);
+        assertEquals(mergedCustom.getData(), "data1");
+    }
+
+    public void testMergeCustomMetaData() {
+        Map<String, MetaData.Custom> mergedCustoms =
+                TribeService.mergeChangedCustomMetaData(Collections.singleton(MergableCustomMetaData1.TYPE),
+                        s -> Arrays.asList(new MergableCustomMetaData1("data1"), new MergableCustomMetaData1("data2")));
+        TestCustomMetaData mergedCustom = (TestCustomMetaData) mergedCustoms.get(MergableCustomMetaData1.TYPE);
+        assertThat(mergedCustom, instanceOf(MergableCustomMetaData1.class));
+        assertNotNull(mergedCustom);
+        assertEquals(mergedCustom.getData(), "data2");
+    }
+
+    public void testMergeMultipleCustomMetaData() {
+        Map<String, List<MetaData.Custom>> inputMap = new HashMap<>();
+        inputMap.put(MergableCustomMetaData1.TYPE,
```

Comment:
the `mergeChangedCustomMetaData` never gets null values for removed customs, instead they are not supplied by the `customMetaDataByTribeNode` function at all when a mergable custom metadata has been removed. I added tests for removing customs in `TribeIT` (`testMergingRemovedCustomMetaData` and `testMergingMultipleCustomMetaData`)